src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -442,8 +442,16 @@ private void updateEventTriggeredFleetStatusData(DeploymentInformation deploymen
                 return;
             }
 
-            // remove any component from unchanged status component list if it's in updatedGreengrassServiceSet
             if (deploymentInformation != null && deploymentInformation.getUnchangedRootComponents() != null) {
+                // for a deployment-triggered FSS update, if any component is NEW state when deployment finishes
+                // (e.g. on-demand lambdas), include them in the updated gg service set because they should be also
+                // installed by this deployment
+                kernel.orderedDependencies().forEach(greengrassService -> {
+                    if (greengrassService.inState(State.NEW) && !isSystemLevelService(greengrassService)) {
```

Comment:
why exclude system? Wouldn't they have this same problem?

Reply:
updated